lib/wordcounter.rb

```diff
@@ -0,0 +1,93 @@
+require 'delegate'
+
+class WordHistogram < SimpleDelegator
+  def initialize(*args)
+    super(Hash.new(0, *args))
+  end
+
+  def to_pretty_s
+    word_freq_pairs = self.sort_histogram
+
+    word_lengths = word_freq_pairs.map { |word, _| word }.map &:length
+    max_length = word_lengths.max
+
+    self.to_table_s(word_freq_pairs, middle_column_width: max_length)
+  end
+
+  protected
```

Comment:
Thanks @mikegee :+1:! It took me too long to catch on, after this exercise and realizing I didn't totally get the semantics I had read some more articles on `self` semantics but I probably should come back to this and push my changes post-review.